Vienna/Sources/Database/Database.m

```diff
@@ -1505,9 +1507,10 @@ -(BOOL)addArticle:(Article *)article toFolder:(NSInteger)folderID
     }
 
     // Dates are stored as time intervals
-    NSTimeInterval interval = articleDate.timeIntervalSince1970;
-    NSTimeInterval createdInterval = article.createdDate.timeIntervalSince1970;
-    
+    NSTimeInterval lastUpdateIntervalSince1970 = lastUpdate.timeIntervalSince1970;
+    NSTimeInterval publicationIntervalSince1970 = 
+        (article.publicationDate == nil ? [NSDate date] : article.publicationDate).timeIntervalSince1970;
```

Comment:
No need to test on `nil` here, as `article.publicationDate` has already been set a few lines ago.